src/dataflow/source/kafka.rs

```diff
@@ -257,17 +255,24 @@ struct DataPlaneInfo {
     /// Kafka consumer for this source
     consumer: Arc<BaseConsumer<GlueConsumerContext>>,
     /// List of consumers. A consumer should be assigned per partition to guarantee fairness
-    partitions: VecDeque<PartitionConsumer>,
+    partition_consumers: VecDeque<PartitionConsumer>,
+    /// The current number of partitions. If needs_refresh is set to false, then
+    /// current_partition_count == expected_partition_count == consumers.len() == partition_metadata.len()
+    current_partition_count: i32,
```

Comment:
It removes the number of conversations I have to do (the next "pid" is always current_partition_count + 1). It's also the maximum number of partitions a Kafka topic can have (because of the pid representation)